pkg/handler/eventhandler.go

```diff
@@ -64,70 +65,70 @@ type EventHandler TypedEventHandler[client.Object]
 // Most users shouldn't need to implement their own TypedEventHandler.
 //
 // TypedEventHandler is experimental and subject to future change.
-type TypedEventHandler[T any] interface {
+type TypedEventHandler[object any, request comparable] interface {
```

Comment:
The main annoyance I found with this change for standard k8s controllers is this, when using a typed eventhandler to have the added typesafety in there it is now also needed to pass the request type. Removing it would mean it is impossible to use a custom request type with `source.Kind` though, which is for example useful when having watches in different clusters